### ISSUE TITLE
False positive in the optionalfields linter

### ISSUE THREAD

**Issue by fabriziopandini** [kubernetes-sigs/kube-api-linter#132]
In Cluster API KAL is reporting

```
api/ipam/v1beta2/ipaddressclaim_types.go:69:2: optionalfields: field AddressRef is optional and has a valid zero value ({}), but the validation is not complete (e.g. min properties/adding required fields). The field should be a pointer to allow the zero value to be set. If the zero value is not a valid use case, complete the validation and remove the pointer. (kubeapilinter)
        AddressRef IPAddressReference `json:"addressRef,omitempty,omitzero"`
```

However the IPAddressReference does not have a valid zero value:

```
// IPAddressReference is a reference to an IPAddress.
type IPAddressReference struct {
	// name of the IPAddress.
	// name must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character.
	// +required
	// +kubebuilder:validation:MinLength=1
	// +kubebuilder:validation:MaxLength=253
	// +kubebuilder:validation:Pattern=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
	Name string `json:"name,omitempty"`
}
```

From a quick experiment, the optionalfields linter is getting tricked by the `omitempty` on the required field (if I remove the omitempty, the issue goes away)

**Comment by fabriziopandini** [comment]
Oh sorry, I forgot to add this
I was working on the required-fields-v2 from your fork 😅

**Comment by sbueringer** [comment]
I think I get the same with `v0.0.0-20250723124831-1b29e82a0f55`

**Comment by sbueringer** [comment]
Okay looks like I"m only getting it with the old version ("v0.0.0-20250723124831-1b29e82a0f55"), but not with your branch. At least on https://github.com/kubernetes-sigs/cluster-api/pull/12545

**Comment by sbueringer** [comment]
I think so

**Comment by fabriziopandini** [comment]
Thanks for looking into it!
